x-pack/plugins/infra/public/pages/metrics/hosts/components/cloud_provider_icon_with_title.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiIcon, EuiText } from '@elastic/eui';
+import React from 'react';
+
+const cloudIcons: Record<string, string> = {
+  gcp: 'logoGCP',
+  aws: 'logoAWS',
+  azure: 'logoAzure',
+  unknownProvider: 'cloudSunny',
+};
+
+export const CloudProviderIconWithTitle = ({
+  provider,
+  title,
+}: {
+  provider?: string | null;
+  title: string;
+}) => {
+  return (
+    <EuiFlexGroup
+      alignItems="center"
+      className="eui-textTruncate"
+      gutterSize="s"
+      responsive={false}
+    >
+      <EuiFlexItem grow={false}>
+        <EuiIcon
+          type={provider ? cloudIcons[provider] : cloudIcons.unknownProvider}
```

Comment:
**nit**: If the provider passed exists but doesn't match anything into the `cloudIcons` map, it will render nothing instead of using the `cloudSunny` icon as a fallback.
Switching to a `(provider && cloudIcons[provider]) || cloudIcons.unknownProvider` conditional should cover the case of the provider exists and has a related entry in the map, otherwise it fallbacks to `cloudIcons.unknownProvider`

Reply:
We should have the cloud icon but that's a good point. It could happen that it's a new cloud provider or a missing one and we should show the fallback for this case. I changed it, thanks for the suggestion!